service/cache/fifo.go

```diff
@@ -1,5 +1,33 @@
 package cache
 
+import (
+	"github.com/karnadattasai/Cache-Go/service/list"
+)
+
 // cacheFIFO hold data structures that implements FIFO
 type cacheFIFO struct {
+	cacheList         list.List
+	keyNodePointerMap map[int]*list.Node
+}
+
+func (c *cacheFIFO) Read(key int) int {
+	if node, ok := c.keyNodePointerMap[key]; ok {
+		return node.P.Value
+	}
+	return -1
+}
+
+func (c *cacheFIFO) Write(key int, value int) {
+	// If given key is already present update the value
```

Comment:
Now I am actually getting confused. How is FIFO different from LRU in your case ??

Reply:
When there is no space in the cache, we will remove the least recently used element in case of LRU  cache. So when ever an element is referenced( here we are only using Write() for this purpose) it will be moved to the end of the list as it will be the most recently used. And in the front of the list we will have least recently used which will be removed to if there is no space.

In case of FIFO cache, when there is no space in the cache, we will remove the first element that entered the cache. When we are referencing (i.e., Write()), we simply update the value and will not move the element to any where like we did in LRU to maintain FIFO. So there will be no c.cacheList.Moveback() like in LRU.

Reply:
Ohh did not see this. Got it. Thanks for the explanation.